types/onesignal-cordova-plugin/index.d.ts

```diff
@@ -1,205 +1,364 @@
-        adm_large_icon?: string;
-        adm_big_picture?: string;
-        chrome_icon?: string;
-        chrome_big_picture?: string;
-        chrome_web_icon?: string;
-        firefox_icon?: string;
-        url?: string;
-        send_after?: string;
-        delayed_option?: string;
-        delivery_time_of_day?: string;
-        android_led_color?: string;
-        android_accent_color?: string;
-        android_visibility?: number;
-        content_available?: boolean;
-        amazon_background_data?: boolean;
-        template_id?: string;
-        android_group?: string;
-        android_group_message?: any;
-        adm_group?: string;
-        adm_group_message?: any;
-        ttl?: number;
-        priority?: number;
-        ios_category?: string;
-    }
-
-    interface OSNotificationPayload {
-        notificationID: string;
-        title: string;
-        body: string;
-        additionalData?: any;
-        smallIcon?: string;
-        largeIcon?: string;
-        bigPicture?: string;
-        smallIconAccentColor?: string;
-        launchURL?: string;
-        sound: string;
-        ledColor?: string;
-        lockScreenVisibility?: OSLockScreenVisibility;
-        groupKey?: string;
-        groupMessage?: string;
-        actionButtons: OSActionButton[];
-        fromProjectNumber?: string;
-        backgroundImageLayout?: OSBackgroundImageLayout;
-        priority?: number;
-        rawPayload: string;
-    }
-
-    interface OSActionButton {
-        id: string;
-        text: string;
-        icon: string;
-    }
-
-    interface OSPermissionState {
-        hasPrompted: boolean;
-        status: OSNotificationPermission;
-    }
-
-    interface OSSubscriptionState {
-        subscribed: boolean;
-        userSubscriptionSetting: any;
-        userId: any;
-        pushToken: any;
-    }
-
-    interface OSPermissionSubscriptionState {
-        permissionStatus: OSPermissionState;
-        subscriptionStatus: OSSubscriptionState;
-    }
-
-    interface OSBackgroundImageLayout {
-        image: string;
-        titleTextColor: string;
-        bodyTextColor: string;
-    }
-
-    interface OSNotificationOpenedResult {
-        action: {
-            type: OSActionType;
-            actionID?: string;
-        };
-        notification: OSNotification;
-    }
-
-    const enum OSLockScreenVisibility {
-        Public = 1,
-        Private = 0,
-        Secret = -1
-    }
-
-    const enum OSDisplayType {
-        None = 0,
-        InAppAlert = 1,
-        Notification = 2
-    }
-
-    const enum OSActionType {
-        Opened = 0,
-        ActionTake = 1
-    }
-
-    const enum OSInFocusDisplayOption {
-        None = 0,
-        InAppAlert = 1,
-        Notification = 2
-    }
-
-    const enum OSNotificationPermission {
-        NotDetermined = 0,
-        Authorized = 1,
-        Denied = 2,
-    }
+	interface OneSignalCordovaPlugin {
+		OSInFocusDisplayOption: { None: 0; InAppAlert: 1; Notification: 2 };
```

Comment:
Are these needed in addition to the enums defined below?

Reply:
Yes, I think so. Enums are actually defined on SDK object and I cannot find better way of writing this. Setting the property as Enum will force it to be `number` in enums range. As far as I know, `enum`s in `d.ts` files doest not create enum object thus are missing in runtime, since the enums are `const` values should be inlined in `ts` files and it probably works, but we are transpiling our application with *babel* and we are using TS just for type checking, so inlining does not work for us.